### ISSUE TITLE
[a11y]: Button: ghost kind in high contrast mode needs higher contrast ratio

### ISSUE THREAD

**Issue by kinueng** [carbon-design-system/carbon#9903]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

Windows

### Package version

latest

### Automated testing tool and ruleset

Windows 10 High Contrast Mode

### Description

The ghost style button needs to be easily visible in high contrast mode.

### WCAG 2.1 Violation

Success Criterion 1.4.3: Contrast (Minimum)

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-button--playground
![image](https://user-images.githubusercontent.com/31117513/137990770-ae7a9c13-836f-4d73-9fd7-a521cd0c2bfc.png)

### Steps to reproduce

1. Turn on Windows 10 high contrast mode, [instructions](https://support.microsoft.com/en-us/windows/turn-high-contrast-mode-on-or-off-in-windows-909e9d89-a0f9-a3a9-b993-7a6dcee85025)
2. Visit https://react.carbondesignsystem.com/?path=/story/components-button--playground
3. Observe the ghost kind button is hard to see due to low color contrast

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
